1.8/gitlab/README.md

```diff
@@ -0,0 +1,190 @@
+```
+
+7. Finally, let's create a simple Jenkinsfile. This is a new way to programmatically define your builds that was released as part of [the Pipeline feature](https://jenkins.io/doc/pipeline/) of Jenkins 2.0. This will pull from the GitLab instance you've just set up, build the Docker image, push it to the GitLab registry, and then trigger a deployment using [the Marathon plugin](https://github.com/jenkinsci/marathon-plugin) for Jenkins:
+
+```
+nano Jenkinsfile
+```
+
+8. Paste the following Groovy script into it. Note that this won't work on an Enterprise DC/OS cluster without saving a token credential and setting the `credentialsId` within the Marathon deploy step:
+
+```
+def gitCommit() { sh "git rev-parse HEAD > GIT_COMMIT" def gitCommit = readFile('GIT_COMMIT').trim() sh "rm -f GIT_COMMIT" return gitCommit }
+
+node { // Checkout source code from Git stage 'Checkout' checkout scm
+
+  // Build Docker image
+  stage 'Build'
+  sh "docker build -t gitlab-test.mesosphere.com:50000/root/gitlab-demo:${gitCommit()} ."
+
+  // Log in and push image to GitLab
+  stage 'Publish'
+  withCredentials(
+      [[
+          $class: 'UsernamePasswordMultiBinding',
+          credentialsId: 'gitlab',
+          passwordVariable: 'GITLAB_PASSWORD',
+          usernameVariable: 'GITLAB_USERNAME'
+      ]]
+  ) {
+      sh "docker login -u ${env.GITLAB_USERNAME} -p ${env.GITLAB_PASSWORD} -e demo@mesosphere.com gitlab-test.mesosphere.com:50000"
+      sh "docker push gitlab-test.mesosphere.com:50000/root/gitlab-demo:${gitCommit()}"
+  }
+
+
+  // Deploy
+  stage 'Deploy'
+
+  marathon(
+      url: 'http://marathon.mesos:8080',
+      forceUpdate: false,
+      filename: 'marathon.json',
+      appid: 'nginx',
+      docker: "gitlab-test.mesosphere.com:50000/root/gitlab-demo:${gitCommit()}".toString()
+  )
+
+}
+```
+
+9. Now that we've got our four files, add, commit and push them to the repository:
+
+```
+git add *
+git commit -m "Inital commit."
+git push
+```
+
+
+## Set up the Pipeline
+
+Now that we've got the repository set up, we need to set up a job in Jenkins that runs on any change to our repository.
+
+1. Navigate to Jenkins. First we'll add the GitLab username and password. For now, we'll re-use the root user, but you can create new users as you see fit. Make sure to call these credentials `gitlab` as referenced in the Jenkinsfile:
+
+![Add GitLab credentials](img/credentials.png)
+
+- Next, go back to the Jenkins homepage and and click on New Item. We'll create a new "Pipeline" job:
+
+![Create New Item](img/new-item.png)
+
+- For this simple example, we'll just select `"Poll SCM" and set the schedule to * * * * *`. This asks us to poll every minute, which might be inefficient for large installations:
+
+![Poll SCM](img/poll-scm.png)
+
+- Next, change the Pipeline definition to use "Pipeline script from SCM" and configure the demo repository:
+
+![Pipeline Configuration](img/pipeline.png)
```

Comment:
Need to update this step since it turns out you need credentials. 

`Next, change the Pipeline definition to use "Pipeline script from SCM". Configure the demo repository and select the credentials you saved earlier from the Credentials dropdown:`

Here's an updated image:
![image](https://cloud.githubusercontent.com/assets/355705/21167259/2497c506-c160-11e6-905a-3ea4aa06d107.png)